### PR TITLE
[SPARK-7036][MLLIB] ALS.train should support DataFrames in PySpark

### PR DESCRIPTION
**Pull request by mengxr**
SchemaRDD works with ALS.train in 1.2, so we should continue support DataFrames for compatibility. @coderxiang

### CODE REVIEW

**Comment by mengxr** [comment]
test this please

File: python/pyspark/mllib/recommendation.py [hunk 1]
```diff
@@ -125,13 +131,20 @@ class ALS(object):
...
     @classmethod
     def _prepare(cls, ratings):
-        assert isinstance(ratings, RDD), "ratings should be RDD"
+        if isinstance(ratings, RDD):
+            pass
+        elif isinstance(ratings, DataFrame):
+            ratings = ratings.rdd
+        else:
+            raise TypeError("Ratings should be represented by either an RDD or a DataFrame, "
+                            "but got %s." % type(ratings))
```
Comment (coderxiang) [inline python/pyspark/mllib/recommendation.py]: a minor issue: maybe better to use "%s is provided"

**Comment by coderxiang** [comment]
LGTM.

**Comment by mengxr** [comment]
Merged into master and branch-1.3.

**Pull request closed without merging**